code/modules/mob/living/basic/lavaland/bileworm/_bileworm.dm

```diff
@@ -0,0 +1,39 @@
+/mob/living/basic/bileworm
+	name = "bileworm"
+	desc = "Bileworms are dangerous detritivores that attack with the highly acidic bile they produce from consuming detritus."
+	icon = 'icons/mob/lavaland/bileworm.dmi'
+	icon_state = "bileworm"
+	icon_living = "bileworm"
+	icon_dead = "bileworm_dead"
+	mob_biotypes = MOB_BUG
+	maxHealth = 150
+	health = 150
+	butcher_results = list(/obj/item/food/meat/slab/bugmeat = 6)
+	guaranteed_butcher_results = list(/obj/effect/gibspawner/generic)
+
+	//it can't be dragged, just butcher it
+	move_resist = INFINITY
+	//doesn't melee, at all.
+	//or move normally.
+
+	combat_mode = TRUE
+	faction = list("mining")
+
+	ai_controller = /datum/ai_controller/basic_controller/bileworm
+	/// bileworm's spewing ability
+	var/datum/action/cooldown/mob_cooldown/projectile_attack/dir_shots/bileworm/spew_bile
+	/// bileworm's resurfacing ability
+	var/datum/action/cooldown/mob_cooldown/resurface/resurface
+
+/mob/living/basic/bileworm/Initialize(mapload)
+	. = ..()
+	ADD_TRAIT(src, TRAIT_IMMOBILIZED, INNATE_TRAIT)
+	ADD_TRAIT(src, TRAIT_LAVA_IMMUNE, INNATE_TRAIT)
+	ADD_TRAIT(src, TRAIT_ASHSTORM_IMMUNE, INNATE_TRAIT)
+	AddElement(/datum/element/basic_body_temp_sensitive, max_body_temp = INFINITY)
+	spew_bile = new()
+	spew_bile.Grant(src)
+	resurface = new()
+	resurface.Grant(src)
+	ai_controller.blackboard[BB_SPEW_BILE] = spew_bile
+	ai_controller.blackboard[BB_RESURFACE] = resurface
```

Comment:
blackboards clean themselves up.

Reply:
Please do not trust blackboards with anything that's not EXPLICITLY managed by the thing that manages the blackboard. I weakrefed everything a while back and it made me wish for the sweet embrace of death

Reply:
it's very hard to understand what you're asking here Lemon

Reply:
Blackboards used to hold a LOT of hard references to things they had no control/ownership over, and I don't want to have to do [this](https://github.com/LemonInTheDark/tgstation/commit/61da2fb85abc89360831897ffd909d70bbbd30bb) again.
I assume this is fine cause arm is smart, but just in general I wanna be cautious with them. Been misused in the past and I wanna be careful.

Reply:
The issue is about how deleting things blackboards hold could lead to hard dels, not how blackboards themselves clean up. Seeing as these things held by the blackboard have no reference outside of the blackboard, they should delete when the mob deletes just fine. But just in case I'll throw in weakrefs to their behaviors themselves